### PR TITLE
JavaScript realms are used to bypass and eliminate many existing web apps security tools - A problem with a WIP solution

### PR DESCRIPTION
**Pull request by weizman**

### CODE REVIEW

**Comment by dontcallmedom** [comment]
@weizman many thanks for your submission! Unfortunately, due to a low level of submissions within the requested timeline, the Program Committee doesn't feel like it can justify holding a 2 days workshops in the original timeframe of June 7-8. The Program Committee will get back to you both on the content of the submission and a possibly new format/schedule in a couple of weeks.

**Comment by tidoust** [comment]
Hi @weizman,

Thanks again for your submission! The Secure the Web Forward workshop was re-launched as a virtual event on 26-28 September 2023 and the program committee has reviewed position papers that were submitted following the initial call for participation. We are pleased to inform your position statement has been accepted and that we will publish and reference it on the workshop's web site, and use it as input to build the agenda of the workshop.

The workshop will primarily be virtual across three two-hour virtual sessions on the 26th, 27th and 28th of September. We hope that these new dates, pending details on actual times, work for you!

Please note that the virtual sessions are targeted at actively discussing topics. We may briefly ask people to introduce concepts or perspectives but do not plan to have formal presentations of the position papers. Program and logistics should be announced beginning of August.

File: papers/wip.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+## Intro
+
+There are many security tools that try to defend web apps in runtime in the browser by redefining and protecting built in JavaScript APIs.
+The need for such tools grows with the difficulty in securing web apps supply chains which results in having a hard time detecting changes in the supply chain that might cause harm in the victim's browser.
+
+For example, if a malicious code manages to execute within the browser of the victim, it can try accessing some secrets stored by the app in the `localStorage` - therefore we have different security tools which attempt to better control access to such sensitive APIs by redefining their behaviour to be more selective on which code is allowed to access it and which code isn't.
+
+## Problem
+
+however, thanks to JavaScript realms, an attacker can ignore such protection quite easily by simply create a new JavaScript realm and access such sensitive APIs via the new realm's unique set of APIs:
```
Comment (tidoust) [inline papers/wip.md]:
```suggestion
However, thanks to JavaScript realms, an attacker can ignore such protection quite easily by simply creating a new JavaScript realm and accessing such sensitive APIs via the new realm's unique set of APIs:
```

File: papers/wip.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+const stolenSecret = localStorage.secret;
+// bypass security tool easily - access the sensitive API via a new realm
+const stolenSecret = document.body.appendChild(document.createElement('iframe')).contentWindow.localStorage.secret
+```
+
+This effectively eliminates such protections almost completely.
+
+## Solution (WIP)
+
+We currntly are working on the most advanced solution to this problem by trying to provide an easy way to handle creation of new realms in runtime before attackers gain access to those.
```
Comment (tidoust) [inline papers/wip.md]:
```suggestion
We are currently working on the most advanced solution to this problem by trying to provide an easy way to handle creation of new realms in runtime before attackers gain access to those.
```

File: papers/wip.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+## Links to related supporting resources
+
+### Read
+
+* Long term research of realms security https://github.com/weizman/awesome-javascript-realms-security
+* GitHub repo of Snow, the solution we are working on to deal with the problem https://github.com/lavamoat/snow
+
+### Watch
+
+* OWASP AppSec talk about realms security and Snow https://www.youtube.com/watch?v=l2l_qnEhx3M
```
Comment (tidoust) [inline papers/wip.md]:
```suggestion
* [Long term research of realms security](https://github.com/weizman/awesome-javascript-realms-security)
* [GitHub repo of Snow](https://github.com/lavamoat/snow), the solution we are working on to deal with the problem

### Watch

* [OWASP AppSec talk about realms security and Snow](https://www.youtube.com/watch?v=l2l_qnEhx3M)
```

**Pull request merged**